cogs/moderation.py

```diff
@@ -120,6 +196,45 @@ async def promote(self, ctx: Context, member: discord.Member, role: discord.Role
         await dm_embed.set_footer(text=f"Server: {ctx.guild.name}")
         await member.send(embed=dm_embed)
 
+    async def _basic_cleanup_strategy(self, ctx: Context, search: int) -> dict:
+        count = 0
+        async for msg in ctx.history(limit=search, before=ctx.message):
+            if msg.author == ctx.me:
+                await msg.delete()
+                count += 1
+        return {"Bot": count}
+
+    async def _complex_cleanup_strategy(self, ctx: Context, search: int) -> Counter:
+        prefixes = tuple(self.bot.get_guild_prefixes(ctx.guild))  # thanks startswith
+
+        def check(member: discord.Bot) -> bool:
```

Comment:
* Why not just use `member: Bot` instead of using `discord.Bot` which is the same but longer.
* Also why the hell are you setting `member` equal to `Bot` type? name it `bot` so that it's obvious what this parameter is, by naming it `member` everyone would expect a  `discord.Member` not `Bot` ...
* And `check` is an undescriptive name for a function, it's not acceptable, what is it checking?